go/vt/vtgate/vcursor_impl.go

```diff
@@ -420,6 +422,20 @@ func parseDestinationTarget(targetString string, vschema *vindexes.VSchema) (str
 
 func (vc *vcursorImpl) planPrefixKey() string {
 	if vc.destination != nil {
+		switch vc.destination.(type) {
+		case key.DestinationKeyspaceID, key.DestinationKeyspaceIDs:
+			resolved, _, err := vc.ResolveDestinations(vc.keyspace, nil, []key.Destination{vc.destination})
+			if err == nil && len(resolved) > 0 {
+				shards := make([]string, len(resolved))
+				for i := 0; i < len(shards); i++ {
+					shards[i] = resolved[i].Target.GetShard()
+				}
+				sort.Strings(shards)
+				return fmt.Sprintf("%s%sDestinationKeyspaceIDsResolved(%s)", vc.keyspace, vindexes.TabletTypeSuffix[vc.tabletType], strings.Join(shards, ","))
```

Comment:
we do not need to be descriptive in plan key. DestinationKeyspaceIDsResolved(%s) can be just %s

Reply:
got it, agree that the verbosity might take extra memory, will shorten it so the overhead could be reduced.